### PR TITLE
feat: Developed products page UI

### PR DESCRIPTION
**Pull request by MahendraDani**
## Related Issue

Closes: #138 

## Description of Changes

- Developed products Page UI as per the given designs.
- The route of the page is `http://localhost:3000/products`
- The design is fully responsive for mobile, tablet and desktop screens.
## Checklist:

- [x] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [x] I have included comments in areas that may be difficult to understand.
- [x] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

## Screenshots
Please provide any necessary screenshots to illustrate the changes made.

https://github.com/SrijanShovit/carbonops-v2/assets/116940083/bd745b48-808a-4d3d-bd6f-5851bba08720

### CODE REVIEW

**Review by SrijanShovit — COMMENTED** [review]
Hello MahendraDani, thanks for rising a Pull request, your contribution is valuable to us. The maintainers will review this Pull Request and provide feedback as soon as possible. Keep the great work up!<p></p>

**Comment by MahendraDani** [comment]
@shreya024 This is the fresh PR for the issue.
Please review the PR and let me know if there are any changes!

**Comment by SrijanShovit** [comment]
Everything looks good 
Just change $ to rupees😂
@MahendraDani

**Comment by MahendraDani** [comment]
@shreya024 @SrijanShovit  Changed the dollars symbol to ruppee symbol.
Capture_carbonops

File: app/products/page.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import React from "react";
+import Navbar from "@/components/products/Navbar";
+import Image from "next/image";
+import sampleImg from "../../public/images/homepage/documents.png";
```
Comment (shreya024) [inline app/products/page.js:4]: Change Image imports as per #158

File: app/products/page.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      productName: "Product Name 2",
+      description:
+        " Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem quis qui accusamus odit similique, expedita velit at cupiditate dolor iure? Laboriosam repudiandae quod pariatur necessitatibus expedita repellendus quidem cum repellat?",
+      price: "₹150",
+    },
+    {
+      image: sampleImg,
+      productName: "Product Name 3",
+      description:
+        " Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem quis qui accusamus odit similique, expedita velit at cupiditate dolor iure? Laboriosam repudiandae quod pariatur necessitatibus expedita repellendus quidem cum repellat?",
```
Comment (shreya024) [inline app/products/page.js:28]: Resolve vercel Error

**Review by shreya024 — CHANGES_REQUESTED** [review]

**Comment by shreya024** [comment]
@MahendraDani This is the vercel error for your reference

[08:12:47.276] Running "vercel build"
[08:12:47.761] Vercel CLI 31.0.1
[08:12:47.981] Error while parsing config file: "/vercel/path0/package-lock.json"
[08:12:47.986] Error while parsing config file: "/vercel/path0/package-lock.json"
[08:12:47.990] Error while parsing config file: "/vercel/path0/package-lock.json"
[08:12:47.991] Installing dependencies...
[08:12:48.244] yarn install v1.22.17
[08:12:48.300] warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[08:12:48.302] [1/4] Resolving packages...
[08:12:50.218] [2/4] Fetching packages...
[08:13:19.313] [3/4] Linking dependencies...
[08:13:19.316] warning " > bootstrap@5.3.0" has unmet peer dependency "@popperjs/core@^2.11.7".
[08:13:19.317] warning "eslint-config-next > @typescript-eslint/parser > @typescript-eslint/typescript-estree > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
[08:13:19.317] warning " > mdb-react-ui-kit@6.1.0" has unmet peer dependency "@types/react@^18.0.9".
[08:13:19.317] warning " > mdb-react-ui-kit@6.1.0" has unmet peer dependency "@types/react-dom@^18.0.3".
[08:13:19.318] warning " > styled-components@5.3.11" has unmet peer dependency "react-is@>= 16.8.0".
[08:13:49.751] [4/4] Building fresh packages...
[08:13:50.370] success Saved lockfile.
[08:13:50.374] Done in 62.13s.
[08:13:50.404] Detected Next.js version: 13.4.6
[08:13:50.406] Error while parsing config file: "/vercel/path0/package-lock.json"
[08:13:50.407] Running "yarn run build"
[08:13:50.578] yarn run v1.22.17
[08:13:50.601] $ next build
[08:13:51.060] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[08:13:51.061] This information is used to shape Next.js' roadmap and prioritize features.
[08:13:51.061] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[08:13:51.061] https://nextjs.org/telemetry
[08:13:51.061] 
[08:13:51.151] - info Creating an optimized production build...
[08:13:51.292] SyntaxError: Unexpected string in JSON at position 75956
[08:13:51.293]     at JSON.parse (<anonymous>)
[08:13:51.293]     at patchIncorrectLockfile (/vercel/path0/node_modules/next/dist/lib/patch-incorrect-lockfile.js:93:33)
[08:14:00.937] - info Compiled successfully
[08:14:00.939] - info Linting and checking validity of types...
[08:14:02.654] 
[08:14:02.654] Failed to compile.
[08:14:02.655] 
[08:14:02.655] ./app/products/page.js
[08:14:02.655] 45:13  Error: Missing "key" prop for element in iterator  react/jsx-key
[08:14:02.655] 
[08:14:02.655] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[08:14:02.706] error Command failed with exit code 1.
[08:14:02.707] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[08:14:02.721] Error: Command "yarn run build" exited with 1

**Comment by MahendraDani** [comment]
@shreya024 I have used `npm` to install some dependencies . Is that the cause of error?
Do we use `yarn` package manager in our project?

**Comment by shreya024** [comment]
> @shreya024 I have used `npm` to install some dependencies . Is that the cause of error? Do we use `yarn` package manager in our project?

No no npm is the way to go...other errors I've commented like  Missing "key" prop for element in iterator react/jsx-key in ./app/products/page.js

**Comment by MahendraDani** [comment]
@shreya024 I have tried to resolve vercel errors in this [commit](https://github.com/SrijanShovit/carbonops-v2/pull/171/commits/6fe83c96b6f8b9e822621c799af3a80c3a84966c).

If there are still any errors please guide me!

File: app/products/page.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      {/* Organization name */}
+      <div className="w-[95%] mx-auto flex justify-center items-center mt-4 mb-8">
+        <h2 className="text-3xl sm:text-4xl text-[#53c6f2]">
+          Organization Name
+        </h2>
+      </div>
+      <div>
+        {products.map((product) => {
+          return (
+            <div className="mt-4">
```
Comment (shreya024) [inline app/products/page.js:43]: provide a key as id

**Review by shreya024 — CHANGES_REQUESTED** [review]

**Comment by MahendraDani** [comment]
@shreya024 Added a key as `id`

File: app/products/page.js (hunk 3 above)
Comment (MahendraDani) [reply app/products/page.js]: Added key as id

File: app/products/page.js [hunk 4]
```diff
@@ -40,7 +43,7 @@ const page = () => {
       <div>
         {products.map((product) => {
           return (
-            <div className="mt-4">
+            <div className="mt-4" id={product.id}>
```
Comment (shreya024) [inline app/products/page.js:46]: Change line 46 to this <div className="mt-4" key={product.id}>

**Review by shreya024 — CHANGES_REQUESTED** [review]

File: app/products/page.js (hunk 4 above)
Comment (MahendraDani) [reply app/products/page.js:46]: @shreya024 I changed this to line 46 already!

**Comment by shreya024** [comment]
@MahendraDani Change line 46 to this <div className="mt-4" key={product.id}>

**Comment by MahendraDani** [comment]
> @MahendraDani Change line 46 to this `<div className="mt-4" key={product.id}>`

@shreya024 I made the necessary changes , please review the PR

**Review by shreya024 — APPROVED** [review]

**Pull request merged**